### DISCUSSION TITLE
API documentation

### DISCUSSION THREAD

**Discussion by ngelus** [oobabooga/text-generation-webui#2224] [General]
Is there an API documentation for this?
I would like a documentation to integrate a character into a Unity game :D

**Comment by indulgence82** [comment]
For creating a character you have to have the api send the character information in your message prompt. If you want it to have a memory you need to create/send a log in the prompt as well. Personally im still figuring out to get decent responses but basically when it's being used in api your setting up and resending who the ai is acting like and what the conversation is at this point. You also will need to filter the messages because it may try to continue the conversation on behalf of the prompter using this method.

**Reply by ngelus** [reply]
yeah. I get that. The character is already created tho. Now I'm literally just trying to get answers from that character via API.

I'd really just love a proper documentation for this API tho. Just so that I can know what's possible and what's not. You know? :D

**Reply by Katehuuh** [reply]
I did [convert](https://gist.github.com/Katehuuh/21bd0a49a403e6a17e49e8014ec0cd28) api-example.py into C#

**Reply by ngelus** [reply]
> I did [convert](https://gist.github.com/Katehuuh/21bd0a49a403e6a17e49e8014ec0cd28) api-example.py into C#

That is useful. Thank you.

**Reply by emberfish** [reply]
Do you guys know how to pass the character context via API? I tried to find a "character_context" field with no luck.

**Reply by zba** [reply]
There is just "prompt" it contains instruction, history and user input, as i understand characters can't work via api, you just implement them yourself in langchain or other way
